### PR TITLE
Refactor "Doc Preview Cleanup" example workflow

### PR DESCRIPTION
**Pull request by omus**
- Ensure sequential execution to avoid issues with concurrently closed PRs
- Improve commit message for preview deletions
- Fix indentation

### CODE REVIEW

File: docs/src/man/hosting.md [hunk 1]
```diff
@@ -404,16 +409,17 @@ jobs:
...
-            if [ -d "previews/PR$PRNUM" ]; then
+          if [ -d "${preview_dir}" ]; then
               git config user.name "Documenter.jl"
               git config user.email "documenter@juliadocs.github.io"
-              git rm -rf "previews/PR$PRNUM"
-              git commit -m "delete preview"
+              git rm -rf "${preview_dir}"
+              git commit -m "Delete preview for #${PR}"
               git branch gh-pages-new $(echo "delete history" | git commit-tree HEAD^{tree})
               git push --force origin gh-pages-new:gh-pages
```
Comment (omus) [inline docs/src/man/hosting.md:418]: Could be problematic if a new PR pushes it's documentation while this workflow is running.

**Review by odow — APPROVED** [review]
This has happened to me a few times, but I can never be bothered fixing it. I just don't merge PRs in quick succession!

File: docs/src/man/hosting.md [hunk 2]
```diff
@@ -404,16 +409,17 @@ jobs:
...
       - name: Delete preview and history + push changes
         run: |
-            if [ -d "previews/PR$PRNUM" ]; then
+          if [ -d "${preview_dir}" ]; then
               git config user.name "Documenter.jl"
               git config user.email "documenter@juliadocs.github.io"
-              git rm -rf "previews/PR$PRNUM"
-              git commit -m "delete preview"
+              git rm -rf "${preview_dir}"
+              git commit -m "Delete preview for #${PR}"
```
Comment (omus) [inline docs/src/man/hosting.md:416]: This commit message is immediately lost so it doesn't matter what it is
```suggestion
              git commit -m "delete preview"
```

File: docs/src/man/hosting.md [hunk 3]
```diff
@@ -404,16 +409,17 @@ jobs:
...
               git branch gh-pages-new $(echo "delete history" | git commit-tree HEAD^{tree})
               git push --force origin gh-pages-new:gh-pages
-            fi
+          fi
         env:
-            PRNUM: ${{ github.event.number }}
+          preview_dir: previews/PR${{ github.event.number }}
+          PR: ${{ github.event.number }}
```
Comment (omus) [inline docs/src/man/hosting.md:422]:
```suggestion
```

**Comment by mortenpi** [comment]
LGTM to me as well. Is it ready to be merged @omus?

**Comment by omus** [comment]
Ready to merge

**Comment by mortenpi** [comment]
Thanks @omus!

**Pull request merged**